app/models/user.rb

```diff
@@ -253,16 +249,27 @@ def cached_following_organizations_ids
 
   def cached_preferred_langs
     Rails.cache.fetch("user-#{id}-#{updated_at}/cached_preferred_langs", expires_in: 24.hours) do
-      langs = []
-      langs << "en" if prefer_language_en
-      langs << "ja" if prefer_language_ja
-      langs << "es" if prefer_language_es
-      langs << "fr" if prefer_language_fr
-      langs << "it" if prefer_language_it
-      langs
+      preferred_languages_array
     end
   end
 
+  # handles both old (prefer_language_*) and new (Array of language codes) formats
+  def preferred_languages_array
```

Comment:
I moved `preferred_languages_array` out of the decorator because it doesn't make much sense for me to have the cached method version in the model and the uncached one in the decorator.
Probably this should be refactored, now or in the future.